Bio/bgzf.py

```diff
@@ -217,8 +217,11 @@
 Text Mode
 ---------
 
-Like the standard library gzip.open(...), the BGZF code defaults to opening
-files in binary mode.
+Like the standard library gzip.open(...), the BGZF code, with bgzf.open(...),
+defaults to opening files in binary mode. However, the BgzfReader and
+BgzfWriter classes currently default to "r", intupreted as text mode. Note
+that for legacy reasons, gzip.open(..., "r") reads in binary mode, while
+bgzf.open(.., "r") reads in text mode.
```

Comment:
Does it really? :sob:

Reply:
I don't know if it would break things to change the behaviour, making the default always "rb" and interpreting "r" as "rb" (or banning "r" all-together and forcing specifying "rt")

Reply:
This is a bit separate from the other open calls, which is why I was thinking a separate pull request for BGZF would be simpler for reviewing and merging. You could revert the BGZF edits here maybe?

What do you think about a warning in BGZF for just "r" suggesting "rb" or "rt" (better to be explicit; likewise for "w")? And also change any BGZF default of just "r" or "w" to the more explicit form (without actually changing the behaviour)?

Reply:
Ok I reverted the changes for `bgzf.py`-- that can be handled in a separate PR